### PR TITLE
Chore/service certs

### LINKED ISSUE #105: Have terraform setup K8sProvider role
and stop providing user aws keys as part of terraform config, and stop copying user aws keys to the k8s provider vm

From https://cdis.slack.com/files/U6R4GAN0H/F7Y018BQA/Setup_K8s_IAM_Roles_:

Currently we run kube-aws with personal AWS credentials that we upload to the kube provisioner via kube-up.sh:

mkdir -p ~/.aws
...
mv credentials ~/.aws

However - AWS allows us to assign a "role" to an EC2 instance which exposes temporary credentials to applications via a metadata service that the AWS SDK's (and kube-aws I'm pretty sure) hook into if credentials are not otherwise provided:

  http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials
  http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html

For example - kube-aws assigns a role the the k8 nodes, so we can fetch a temporary key for that role by querying a node's metadata service:

$ kubectl exec gdcapi-deployment-3057747319-qpt77 curl http://169.254.169.254/latest/meta-data/iam/security-credentials/planxplanetv1-Nodepool-4VY43DV1ISHF-IAMRoleWorker-158QBVD5WMYA7
 % Total  % Received % Xferd Average Speed  Time  Time   Time Current
                 Dload Upload  Total  Spent  Left Speed
100  894 100  894  0   0 1064k   0 --:--:-- --:--:-- --:--:-- 873k
{
 "Code" : "Success",
 "LastUpdated" : "2017-11-09T22:01:42Z",
 "Type" : "AWS-HMAC",
 "AccessKeyId" :

First, I would like to extend terraform to assign a 'KubeProvider' role to the kube provider EC2 instance with admin permissions, and remove the code that installs the ~/.aws/credentials of whoever ran terraform, so that kube-aws and the aws-cli will run with the 'KubeProvider' role.

### LINKED ISSUE #110: Create Code/Function to Create SSL Certs

### PR DESCRIPTION
**Pull request by frickjack**
## resolve #105 
terraform launches the kube provisioner with an AWS instance profile that maps to a role with less than admin permissions.  kube-aws and the aws cli acquire temporary creds from the AWS metadata service associated with the kube-provisioner's EC2 instance.  We no longer copy ~/.aws/credentials to the kube provisioner

## resolve #110 
the tf_files/configs/kube-certs.sh script runs with kube-services.sh, and can also be run independently at any time to automatically create certificates (via the k8s CA configs in ~/VPC_NAME/credentials) and k8s secrets for k8s services discovered via grep'ing into the services/ folder.  

This patch also updates the various kube/*/*-deployment.yaml files to mount the appropriate SSL secrets under /mnt/ssl.  The kube/README.md has more details

## resolve planx-misc-issue-tracker issue 15 - https://github.com/uc-cdis/planx-misc-issue-tracker/issues/15 
the services/workspace/deploy_workspace.sh script creates
* a k8s namespace (named 'workspace') 
* k8s user (named 'worker') that authenticates via a certificate built from the k8s CA (under ~/VPC_NAME/credentials)
* a role and rolebinding that grants the user permission to deploy services under the namespace
* a kubeconfig_worker with context configured to access the API as the new user
* a kubeconfig_worker.tar.xz suitcase that can be dropped on a VM with a route to the k8s api server to issue kubectl commands with

There's a workspace/README.md with an overview.

### CODE REVIEW

File: tf_files/aws/data.tf [hunk 1]
```diff
@@ -1,2 +1,290 @@
...
+      effect = "Allow"
+      actions = [
+          "ecr:*",
+          "cloudtrail:LookupEvents"
+      ],
+      resources = [ "*" ]
+    }
+
+    statement {
+      actions = [ "ec2:*" ]
```
Comment (zflamig) [inline tf_files/aws/data.tf]: Is this thing above redundant then?

File: tf_files/aws/data.tf [hunk 2]
```diff
@@ -1,2 +1,290 @@
...
+          test = "StringLike"
+          variable = "iam:AWSServiceName"
+          values = [ "ec2scheduled.amazonaws.com" ]
+        }
+    }
+    
+    statement {
+        actions = [
+            "rds:*",
+            "cloudwatch:DescribeAlarms",
```
Comment (zflamig) [inline tf_files/aws/data.tf]: We already allowed cloudwatch:*?

File: tf_files/aws/data.tf [hunk 3]
```diff
@@ -1,2 +1,290 @@
...
+        resources = [ "arn:aws:iam::*:role/aws-service-role/rds.amazonaws.com/AWSServiceRoleForRDS" ]
+        condition = {
+          test = "StringLike"
+          variable = "iam:AWSServiceName"
+          values = [ "rds.amazonaws.com" ]
+        }
+    }
+
+    statement {
+      effect = "Allow"
```
Comment (zflamig) [inline tf_files/aws/data.tf]: Can we get rid of this? Or at least reduce the scope of it? This will allow you to do everything...?

**Comment by zflamig** [comment]
Can you add port 443 to the *-service.yamls in kube/services/* too please?

**Comment by frickjack** [comment]
Thanks, Zac!
* I'll add port 443 to the services.yaml's
* Simplifying and scaling down the provisioner permissions to be just enough to run kube-aws (not terraform):
     https://github.com/kubernetes-incubator/kube-aws/issues/88

**Comment by frickjack** [comment]
Hey Zac,

I just pushed a patch:
* added port 443 to the deployments and services
* cleanup of the provisioner permissions

The permissions on the provisioner are still very broad - basically 'admin', since it gives IAM*, so it's not really any better than what we were doing before (copying up admin credentials.json) except that it gives us a path forward to trim down the permissions on the role (if we update the permissions, then terraform apply updates the inline policy in place).  It's a project to get the right set of permissions for kube-aws - the kube-aws project actually has an open issue:
    https://github.com/kubernetes-incubator/kube-aws/issues/90

Anyway - I'd like to just leave the provisioner permissions the way they are, since it's no worse than what we had been doing, and create a separate issue to narrow down the permissions.   Another option is to wire up 'kube-up.sh' to run 'aws iam delete-role-policy ...' after kube-aws up finishes - which will just delete the permission on the provisioner down to nothing, or put a more limited set of permissions in place after kube-aws has done its thing.  What do you think?

**Pull request merged**